opus/application/static_media/js/opus.js

```diff
@@ -114,227 +141,284 @@ var opus = {
+         */
+
         opus.resultCount = resultCount;
         $("#op-result-count").fadeOut("fast", function() {
             $(this).html(o_utils.addCommas(opus.resultCount)).fadeIn("fast");
             $(this).removeClass("browse_results_invalid");
         });
     },
 
+    //------------------------------------------------------------------------------------
+    // Functions related to nav bar tabs and other menu items
+    //------------------------------------------------------------------------------------
+
     triggerNavbarClick: function() {
+        /**
+         * Simulate a click on the nav bar tab for the current view. This is a
+         * simple way to get the tab to be selected and to execute the associated
+         * event code.
+         */
+
         $('.nav-item a[href="#'+opus.prefs.view+'"]').trigger("click");
     },
 
-    lastBlogUpdate: function() {
+    updateLastBlogDate: function() {
+        /**
+         * Retrieve the date of the last blog update and update the tooltip for
+         * the 'Recent Announcements' nav bar item.
+         */
+
         $.getJSON("/opus/__lastblogupdate.json", function(data) {
             if (data.lastupdate !== null) {
-                let last_update_date = new Date(data.lastupdate);
+                let lastUpdateDate = new Date(data.lastupdate);
                 let today = Date.now();
-                let days = (today - last_update_date.valueOf())/1000/60/60/24;
-                if (days <= 7) {
+                let days = (today - lastUpdateDate.valueOf())/1000/60/60/24;
+                if (days <= 31) { // Show it for a month for infrequent users
                     $(".blogspot img").show();
                 } else {
                     $(".blogspot img").hide();
                 }
-                let pretty_date = last_update_date.toLocaleDateString('en-GB', {year: 'numeric', month: 'long', day: 'numeric'});
-                $("#last_blog_update_date").attr("title", "Blog last updated "+pretty_date);
+                let prettyDate = lastUpdateDate.toLocaleDateString('en-GB',
+                                        {year: 'numeric', month: 'long', day: 'numeric'});
+                $("#op-last-blog-update-date").attr("title", "Blog last updated "+prettyDate);
             } else {
-                $("#last_blog_update_date").attr("title", "");
+                $("#op-last-blog-update-date").attr("title", "");
             }
         });
     },
 
     changeTab: function(tab) {
-        // first hide everything and stop any interval timers
-        $('#search, #detail, #cart, #browse').hide();
+        /**
+         * This is the event handler for the user clicking on one of the main nav
+         * bar tabs (views).
+         */
+
+        // First hide everything and stop any interval timers
+        $("#search, #detail, #cart, #browse").hide();
         o_browse.hideMenu();
 
-        // close any open modals
+        // Close any open modals
```

Comment:
what about the metadata modal? do we need to worry about closing that here as well?  if so, do we need a "closeModals' function?

Reply:
There's a separate handler in browse.js for ESC on the metadata modal and also the select metadata modal. It also does things like stop the range select and, for no obvious reason, close the help pane. I think the right thing to do is to leave the metadata modal and select metadata modal in browse.js because they have other actions associated with them (like asking for confirmation closing the select metadata...eventually), and move the close help pane to opus.js. It's sort of a tricky decision because the confirm modals are scattered all over the place, so it's difficult to have a handler for each namespace separately. Putting them in opus.js seems like the best choice.